NookureChat-API/src/main/java/com/nookure/chat/api/TextUtils.java

```diff
@@ -33,29 +36,41 @@ public static String processPlaceholders(Player player, String message) {
   public static String toMM(String str) {
     if (!str.contains("&") && !str.contains("§")) return str;
 
-    return str
-        .replace("§", "&")
-        .replace("&0", "<reset><black>")
-        .replace("&1", "<reset><dark_blue>")
-        .replace("&2", "<reset><dark_green>")
-        .replace("&3", "<reset><dark_aqua>")
-        .replace("&4", "<reset><dark_red>")
-        .replace("&5", "<reset><dark_purple>")
-        .replace("&6", "<reset><gold>")
-        .replace("&7", "<reset><grey>")
-        .replace("&8", "<reset><dark_grey>")
-        .replace("&9", "<reset><blue>")
-        .replace("&a", "<reset><green>")
-        .replace("&b", "<reset><aqua>")
-        .replace("&c", "<reset><red>")
-        .replace("&d", "<reset><light_purple>")
-        .replace("&e", "<reset><yellow>")
-        .replace("&f", "<reset><white>")
-        .replace("&k", "<obf>")
-        .replace("&l", "<b>")
-        .replace("&m", "<st>")
-        .replace("&n", "<u>")
-        .replace("&r", "<reset>")
-        .replace("&o", "<i>");
+    str = str
+            .replace("§", "&")
+            .replace("&0", "<black>")
+            .replace("&1", "<dark_blue>")
+            .replace("&2", "<dark_green>")
+            .replace("&3", "<dark_aqua>")
+            .replace("&4", "<dark_red>")
+            .replace("&5", "<dark_purple>")
+            .replace("&6", "<gold>")
+            .replace("&7", "<grey>")
+            .replace("&8", "<dark_grey>")
+            .replace("&9", "<blue>")
+            .replace("&a", "<green>")
+            .replace("&b", "<aqua>")
+            .replace("&c", "<red>")
+            .replace("&d", "<light_purple>")
+            .replace("&e", "<yellow>")
+            .replace("&f", "<white>")
+            .replace("&k", "<obf>")
+            .replace("&l", "<b>")
+            .replace("&m", "<st>")
+            .replace("&n", "<u>")
+            .replace("&o", "<i>")
+            .replace("&r", "<reset>");
+
+    Pattern hexPattern = Pattern.compile("&#([A-Fa-f0-9]{6})");
```

Comment:
You can cache the pattern instead of creating a new object all the time

```java
private static final Pattern hexPattern = Pattern.compile("&#([A-Fa-f0-9]{6})");
```
on the top level